src/command/process_update_task.rs

```diff
@@ -110,18 +113,36 @@ impl UpdateProcessor {
         }
     }
 
-    pub async fn process(&self) -> Result<(), BotError> {
+    pub async fn process(&self) -> Result<Option<ScheduleWeatherTask>, BotError> {
         self.send_typing().await?;
 
         if Command::Cancel == self.command {
-            return self.cancel(None).await;
+            self.cancel(None).await?;
+            return Ok(None);
         }
 
         match self.chat.state {
-            ClientState::Initial => self.process_initial().await,
-            ClientState::FindCity => self.process_find_city().await,
-            ClientState::Number => self.process_number().await,
-            _ => self.revert_state().await,
+            ClientState::Initial => {
+                self.process_initial().await?;
+                Ok(None)
+            }
+            ClientState::FindCity => {
+                self.process_find_city().await?;
+                Ok(None)
+            }
+            ClientState::Number => {
+                self.process_number().await?;
+                Ok(None)
+            }
+            ClientState::Time => {
```

Comment:
I think the command to stop receiving updates should also be added.

And maybe commands to check:
- current default city
- current offset
- current scheduled time

Reply:
[Current default city command](https://github.com/pxp9/weather_bot_rust/pull/29/commits/a4002fe5680a21cbd54de8f0c7de0984eafa7c68)

Reply:
Offset is not stored in database so, maybe would we like to store it ?

Reply:
User should have multiple scheduled times, should we limit to 1 scheduled time ?

Reply:
> Offset is not stored in database so, maybe should we store it ?

I think the offset should be stored

> User can have multiple scheduled times, should we limit to 1 scheduled time ?

I think for now it's fine to have one scheduled time